libcontainer/container_linux.go

```diff
@@ -105,8 +105,8 @@ func (c *linuxContainer) Start(process *Process) error {
 		return newSystemError(err)
 	}
 	if err := parent.start(); err != nil {
-		// terminate the process to ensure that it properly is reaped.
-		if err := parent.terminate(); err != nil {
+		// wait for the process to ensure that it properly is reaped.
+		if _, err := parent.wait(); err != nil {
```

Comment:
What if start() fails after a process was spawned successfully, say, if applying cgroups failed? Couldn't this hang forever?